tidy3d/plugins/design/design.py

```diff
@@ -128,52 +322,175 @@ def run_batch(
+                arg_dict[param.name] = param.allowed_values[0]
 
-        # run the functions using the `method.run_batch`
-        fn_args, fn_values, task_ids, batch_data = self.method.run_batch(
-            parameters=self.parameters,
-            fn_pre=fn_pre,
-            fn_post=fn_post,
-            path_dir=path_dir,
-            **batch_kwargs,
-        )
+        # Compute fn_pre
+        pre_out = fn_pre(**arg_dict)
 
-        # store the pre and post functions in a single source code
-        fn_source_pre = self.get_fn_source(fn_pre)
-        fn_source_post = self.get_fn_source(fn_post)
-        fn_source = self._make_batch_fn_source(fn_source_pre, fn_source_post)
+        def _estimate_sim_cost(sim):
+            job = Job(simulation=sim, task_name="estimate_cost")
 
-        # package the result
-        return self._package_run_results(
-            fn_args=fn_args,
-            fn_values=fn_values,
-            fn_source=fn_source,
-            task_ids=task_ids,
-            batch_data=batch_data,
+            estimate = job.estimate_cost()
+            job.delete()  # Deleted as only a test with initial parameters
+
+            return estimate
+
+        if isinstance(pre_out, Simulation):
+            per_run_estimate = _estimate_sim_cost(pre_out)
+
+        elif isinstance(pre_out, Batch):
+            per_run_estimate = pre_out.estimate_cost()
+            pre_out.delete()  # Deleted as only a test with initial parameters
+
+        elif isinstance(pre_out, (list, dict)):
+            # Iterate through container to get simulations and batches and sum cost
+            # Accept list or dict inputs
+
+            if isinstance(pre_out, dict):
+                pre_out = list(pre_out.values())
+
+            sims = []
+            batches = []
+            for value in pre_out:
+                if isinstance(value, Simulation):
+                    sims.append(value)
+                elif isinstance(value, Batch):
+                    batches.append(value)
+
+            calculated_estimates = []
+            for sim in sims:
+                calculated_estimates.append(_estimate_sim_cost(sim))
+
+            for batch in batches:
+                calculated_estimates.append(batch.estimate_cost())
+                batch.delete()  # Deleted as only a test with initial parameters
+
+            if None in calculated_estimates:
+                per_run_estimate = None
+            else:
+                per_run_estimate = sum(calculated_estimates)
+
+        else:
+            raise ValueError("Unrecognised output from pre-function, unable to estimate cost.")
+
+        # Calculate maximum number of runs for different methods
+        run_count = self._get_run_count()
+
+        # For if tidy3d server cannot determine the estimate
+        if per_run_estimate is None:
+            return None
+        else:
+            return round(per_run_estimate * run_count, 3)
+
+    def summary(self, fn_pre: Callable = None):
```

Comment:
`-> None:`

I think if we go with the console.log approach, then this should be called `summarize`, or `print_summary`

it might be nice to get the summary as a `pd.BaseModel`, `dataclass`, or  `dict` as well

Reply:
Changed to output the summary as dict so that the user has access to a the data as a variable. Works well for now, can be expanded in the future.